### PR TITLE
Restructure file data classes

### PR DESCRIPTION
**Pull request by jacobrh91**
@sstevenshang Can you review this PR for me?

I removed the Sparse Matrix implementation fragments, and refactored the parsers and added in a package controlling the data format that will be "plugged in" to the front of each model building program.

### CODE REVIEW

File: SPAEML/src/main/scala/spaeml/SPAEML.scala [hunk 1]
```diff
@@ -135,38 +132,212 @@ object SPAEML {
     writer.close()
   }
 
-}
+/*}
```
Comment (sstevenshang) [inline SPAEML/src/main/scala/spaeml/SPAEML.scala:135]: Remove the commented out class definition here.

File: SPAEML/src/main/scala/spaeml/SPAEML.scala [hunk 2]
```diff
@@ -135,38 +132,212 @@ object SPAEML {
...
+         */
+        if (collections.getSkipped.isEmpty) return bestRegression
+        else return rebuildModel()
+      }
+      else {
+        /*
+         * If any entries were skipped this round, recompute the regression without these terms,
+         *  and then include the new best regression in the next iteration
+         */
+        val newBestReg: OLSRegression =  if (collections.getSkipped.nonEmpty) rebuildModel() else bestRegression
```
Comment (sstevenshang) [inline SPAEML/src/main/scala/spaeml/SPAEML.scala:331]: This function (`performSteps`) is pretty complicated. Doing some logging here will help us visualize the program flow and help debug any problem.

**Review by sstevenshang — APPROVED** [review]
Looks good. Just some general comments. Ready for merge.

I have one comment on documentation: 
- It'd be very helpful to have proper param docs for every function (explaining **what** the function does, what each parameter is, what the output is, and whether or not there is any side-effect). In most cases, this is more useful then comments about **how** a function works.

File: SPAEML/src/main/scala/spaeml/SPAEML.scala (hunk 2 above)
Comment (jacobrh91) [reply SPAEML/src/main/scala/spaeml/SPAEML.scala:331]: I agree with the params comment. I'll work on that in another pull request. And certainly, logging would help us debug that huge piece of logic! I'll get on it.

Comment (sstevenshang) [reply SPAEML/src/main/scala/spaeml/SPAEML.scala:331]: Let's merge this one for now? So I can get rebase my other branch.

**Pull request merged**